sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1263,6 +1264,8 @@ object SQLContext {
    */
   @transient private val instantiatedContext = new AtomicReference[SQLContext]()
 
+  @transient private val sqlListener = new AtomicReference[SQLListener]()
```

Comment:
Many unit tests use `sqlContext.listener`. Can you please suggest how to update the unit tests if we changed to use an AtomicBoolean?